### ISSUE TITLE
Using enum inside export value gives error in 3.0.1.

### ISSUE THREAD

**Issue by OGMP-Gyrth** [godotengine/godot#17025]
OS: Ubuntu Linux 17.10
Godot version: Godot_v3.0.1-stable_x11.64

I use my own enum inside a script that gets loaded into a bunch other scripts. So I use this file as my source for global values. However in the latest version of Godot this doesn't seem to work anymore. I don't know if this is me using it wrong or this is a bug, please tell me. :smile: 

![image](https://user-images.githubusercontent.com/14081613/36647727-3e219be4-1a8a-11e8-87bc-cef12d00d5a9.png)

![image](https://user-images.githubusercontent.com/14081613/36647784-ed8c806c-1a8a-11e8-82ad-882ca8191e7f.png)

Example project:
[BugReport.zip](https://github.com/godotengine/godot/files/1756833/BugReport.zip)

**Comment by hpvb** [comment]
Hi! Thanks for the bugreport and sorry about this. This is exactly the sort of breakage I'm trying to avoid in the stable series. I'll figure out why this is happening and make sure this gets fixed in 3.0.2 or at least documented if this is somehow unavoidable.

**Comment by hpvb** [comment]
So the good news is that it appears to be cosmetic. The functionality still works so it's a problem with the editor. It won't actually break your game. So this can be fixed in 3.0.2. If this is too annoying to work with please revert to 3.0 for the moment :-/

**Comment by hpvb** [comment]
Caused by b7faa76485be1cfe368d24e4fe30adf252beedbe

**Comment by x1212** [comment]
Oh, I'll see if I have time to have a look at what my commit broke when I get home.
My guess right now would be that due to GDScript no longer forgetting if it was called for autocompletion/validation some bug elsewhere was exposed but I can't say for sure without doing some debugging.

**Comment by hpvb** [comment]
@x1212 Instead of calling clear() in _parse_bytecode() and parse() duplicating all of clear() except clearing the already data seems to solve this issue. 

```c++
	completion_type = COMPLETION_NONE;
	completion_node = NULL;
	completion_class = NULL;
	completion_function = NULL;
	completion_block = NULL;
	current_block = NULL;
	current_class = NULL;

	completion_found = false;
	rpc_mode = ScriptInstance::RPC_MODE_DISABLED;

	current_function = NULL;

	validating = false;
	for_completion = false;
	error_set = false;
	tab_level.clear();
	tab_level.push_back(0);
	error_line = 0;
	error_column = 0;
	pending_newline = -1;
	parenthesis = 0;
	current_export.type = Variant::NIL;
	error = "";
```
instead of clear() I also put the clear() back in _parse() as I couldn't quite figure out why it had to be removed.

**Comment by x1212** [comment]
You can also fix this by replacing the `!validating` with `true` in line 456 of modules/gdscript/gdscript_parser.cpp ... so I guess the problem is that now that the validating member works, some optimization in the parser supposed to make validating faster breaks validating?
The part that isn't executed now is supposed to load preloaded scripts ... and the errormessage is caused by the parser not understanding that target_types is a const member of the preloaded const script.

**Comment by x1212** [comment]
Just saw that 3.0.2 already got it's first rc ...
I wanted to look for a better solution but I guess I'll create a pullrequest today that removes that !validating condition for parsing preloaded scripts.

**Comment by OGMP-Gyrth** [comment]
Tested it in 3.0.2 and it seems to be fixed! Thanks!

**Comment by akien-mga** [comment]
It's still valid in the master branch though.

**Comment by DleanJeans** [comment]
Why is this merged into the 3.1 master branch? I'm waiting for so many new features and bug fixes in 3.1.
